shared-bindings/_protomatter/Protomatter.c

```diff
@@ -0,0 +1,288 @@
+        args[ARG_width].u_int,
+        args[ARG_bit_depth].u_int,
+        rgb_count, rgb_pins,
+        addr_count, addr_pins,
+        clock_pin, latch_pin, oe_pin,
+        args[ARG_doublebuffer].u_bool,
+        framebuffer, NULL);
+
+    claim_and_never_reset_pins(args[ARG_rgb_list].u_obj);
+    claim_and_never_reset_pins(args[ARG_addr_list].u_obj);
+    claim_and_never_reset_pin(args[ARG_clock_pin].u_obj);
+    claim_and_never_reset_pin(args[ARG_oe_pin].u_obj);
+    claim_and_never_reset_pin(args[ARG_latch_pin].u_obj);
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+//|   .. method:: deinit
+//|
+//|     Free the resources (pins, timers, etc.) associated with this
+//|     protomatter instance.  After deinitialization, no further operations
+//|     may be performed.
+//|
+STATIC mp_obj_t protomatter_protomatter_deinit(mp_obj_t self_in) {
+    protomatter_protomatter_obj_t *self = (protomatter_protomatter_obj_t*)self_in;
+    common_hal_protomatter_protomatter_deinit(self);
+    return mp_const_none;
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(protomatter_protomatter_deinit_obj, protomatter_protomatter_deinit);
+
+static void check_for_deinit(protomatter_protomatter_obj_t *self) {
+    if (!self->core.rgbPins) {
+        raise_deinited_error();
+    }
+}
+
+//|   .. attribute:: paused
+//|
+//|     When paused, the matrix is not driven and all its LEDs are unlit.
+//|
+STATIC mp_obj_t protomatter_protomatter_get_paused(mp_obj_t self_in) {
+    protomatter_protomatter_obj_t *self = (protomatter_protomatter_obj_t*)self_in;
+    check_for_deinit(self);
+    return mp_obj_new_bool(common_hal_protomatter_protomatter_get_paused(self));
+}
+MP_DEFINE_CONST_FUN_OBJ_1(protomatter_protomatter_get_paused_obj, protomatter_protomatter_get_paused);
+
+STATIC mp_obj_t protomatter_protomatter_set_paused(mp_obj_t self_in, mp_obj_t value_in)  {
+    protomatter_protomatter_obj_t *self = (protomatter_protomatter_obj_t*)self_in;
+    check_for_deinit(self);
+    bool paused = mp_obj_is_true(value_in);
+    common_hal_protomatter_protomatter_set_paused(self, paused);
+
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_2(protomatter_protomatter_set_paused_obj, protomatter_protomatter_set_paused);
+
+const mp_obj_property_t protomatter_protomatter_paused_obj = {
+    .base.type = &mp_type_property,
+    .proxy = {(mp_obj_t)&protomatter_protomatter_get_paused_obj,
+              (mp_obj_t)&protomatter_protomatter_set_paused_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+//|   .. method:: swapbuffers()
+//|
+//|     Transmits the color data in the buffer to the pixels so that they are shown.
+//|
+//|     The data in the buffer must be in "RGB565" format.  This means
+//|     that it is organized as a series of 16-bit numbers where the highest 5
+//|     bits are interpreted as red, the next 6 as green, and the final 5 as
+//|     blue.  The object can be any buffer, but `array.array` and `ulab.array`
+//|     objects are most often useful.
```

Comment:
I don't think this is the right place for this because it isn't taking a buffer in.